tests/modules/images/webp-uploads/load-tests.php

```diff
@@ -608,4 +608,27 @@ public function it_should_replace_the_featured_image_to_web_p_when_requesting_th
 		$this->assertStringContainsString( '.webp', $featured_image );
 		$this->assertStringNotContainsString( '.jpeg', $featured_image );
 	}
+
+	/**
+	 * Prevent replacing an image if image was uploaded via external source or plugin.
+	 *
+	 * @group webp_uploads_update_image_references
+	 *
+	 * @test
+	 */
+	public function it_should_prevent_replacing_an_image_uploaded_via_external_source() {
+		remove_all_filters( 'webp_uploads_pre_replace_additional_image_source' );
+
+		add_filter(
+			'webp_uploads_pre_replace_additional_image_source',
+			function() {
+				return TESTS_PLUGIN_DIR . '/tests/testdata/modules/images/leafs.jpg';
```

Comment:
While this is just a test, what the code here is doing is not correct based on what the filter is for: The filter receives the full image HTML tag, but here the code is replacing it with just a URL to an image. Let's make sure we use a proper example and instead return an **altered `img` tag** that e.g. includes this image URL as its `src`.

Reply:
yes, I have fixed it with `img` tag instead.